tests/suites/test_suite_ssl.function

```diff
@@ -3454,10 +3541,11 @@ exit:
     mbedtls_ssl_transform_free( &t1 );
 
     mbedtls_free( buf );
+    USE_PSA_DONE( );
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_AES_C:MBEDTLS_SSL_PROTO_TLS1_2 */
+/* BEGIN_CASE depends_on:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_AES_C:MBEDTLS_SSL_PROTO_TLS1_2:!MBEDTLS_USE_PSA_CRYPTO */
```

Comment:
This test function calls `mbedtls_cipher_crypt(&t0.cipher_ctx_enc, ...)`. This field (`mbedtls_ssl_transform.cipher_ctx_enc`) is not available in psa builds. This is the reason why I added `!MBEDTLS_USE_PSA_CRYPTO` dependency. 

Maybe we need `mbedtls_cipher_crypt()` version for psa.

Reply:
Oh, OK, the test's body uses `mbedtls_cipher_crypt()` in order to prepare (invalid) inputs to `mbedtls_ssl_decrypt_buf()`, and it's taking the keys from the context. This test is quite important because it ensures that invalid records are rejected as they should. We don't have enough negative testing in SSL in general, so we really don't want less of it with `USE_PSA_CRYPTO`.

So, I think we need to adapt that test so that it works with `USE_PSA_CRYPTO` too. As you said, we probably want to write a simple wrapper function that calls either `mbedtls_cipher_crypt()` or the equivalent sequence of `psa_cipher_xxx()` functions. This should be pretty similar to what you already did in `ssl_encrypt_buf()` in the CBC case, specifically [the block on lines 1480-1505](https://github.com/ARMmbed/mbedtls/pull/5434/files#diff-67127675c5612072a0cb4c12b6dd40b0cfdcc6cdb23839f48ce9e3a2257e77e7R1479) I think.

Reply:
Ok I'll check this.

Reply:
I could add function: `psa_cipher_crypt(mbedtls_ssl_transform *transform, ....)` (first param is pointer to transform structure). Seems that I will have access to everything what is needed to perform equivalent sequence of `psa_cipher_xxx()` calls, but I don't see how to determine the operation type. Current code uses  `ctx->operation` which can't be used in case of psa now.

Reply:
I simply added helper function in the test for encrypt using psa crypto. This is sufficient and  it works, but some of the test cases (`3DES`) are failing:
```
Decrypt CBC !EtM, 3DES MD5 !trunc, empty plaintext, minpad ........ 
mbedtls_ssl_cipher_to_psa: -134 (not supported) FAILED
  ret == 0
  at line 3634, suites/test_suite_ssl.function
```

They are failing because translation to psa fails for `MBEDTLS_CIPHER_DES_EDE3_CBC` cipher. Do we need these test cases?